openedx/core/djangoapps/theming/helpers.py

```diff
@@ -52,3 +71,341 @@ def get_themed_template_path(relative_path, default_path, **kwargs):
+    """
+    from edxmako.middleware import REQUEST_CONTEXT
+    request = getattr(REQUEST_CONTEXT, 'request', None)
+    if not request:
+        return None
+
+    # if hostname is not valid
+    if not all((isinstance(request.get_host(), basestring), is_valid_hostname(request.get_host()))):
+        return None
+
+    return request.site
+
+
+def get_current_site_theme_dir():
+    """
+    Return theme directory for the current site.
+
+    Example:
+        >> get_current_site_theme_dir()
+        'red-theme'
+
+    Returns:
+         (str): theme directory for current site
+    """
+    site = get_current_site()
+    if not site:
+        return None
+    site_theme_dir = cache.get(get_site_theme_cache_key(site))
+
+    # if site theme dir is not in cache and comprehensive theming is enabled then pull it from db.
+    if not site_theme_dir and is_comprehensive_theming_enabled():
+        site_theme = site.themes.first()  # pylint: disable=no-member
+        if site_theme:
+            site_theme_dir = site_theme.theme_dir_name
+            cache_site_theme_dir(site, site_theme_dir)
+    return site_theme_dir
+
+
+def get_project_root_name():
+    """
+    Return root name for the current project
+
+    Example:
+        >> get_project_root_name()
+        'lms'
+        # from studio
+        >> get_project_root_name()
+        'cms'
+
+    Returns:
+        (str): component name of platform e.g lms, cms
+    """
+    root = Path(settings.PROJECT_ROOT)
+    if root.name == "":
+        root = root.parent
+    return root.name
+
+
+def get_base_theme_dir():
+    """
+    Return base directory that contains all the themes.
+
+    Example:
+        >> get_base_theme_dir()
+        '/edx/app/edxapp/edx-platform/themes'
+
+    Returns:
+         (Path): Base theme directory path
+    """
+    themes_dir = settings.COMPREHENSIVE_THEME_DIR
+    if not isinstance(themes_dir, basestring):
+        raise ImproperlyConfigured("COMPREHENSIVE_THEME_DIR must be a string.")
+    return Path(themes_dir)
+
+
+def is_comprehensive_theming_enabled():
+    """
+    Returns boolean indicating whether comprehensive theming functionality is enabled or disabled.
+    Example:
+        >> is_comprehensive_theming_enabled()
+        True
+
+    Returns:
+         (bool): True if comprehensive theming is enabled else False
+    """
+    return True if settings.COMPREHENSIVE_THEME_DIR else False
+
+
+def get_site_theme_cache_key(site):
+    """
+    Return cache key for the given site.
+
+    Example:
+        >> site = Site(domain='red-theme.org', name='Red Theme')
+        >> get_site_theme_cache_key(site)
+        'theming.site.red-theme.org'
+
+    Parameters:
+        site (django.contrib.sites.models.Site): site where key needs to generated
+    Returns:
+        (str): a key to be used as cache key
+    """
+    cache_key = "theming.site.{domain}".format(
+        domain=site.domain
+    )
+    return cache_key
+
+
+def is_valid_hostname(hostname):
```

Comment:
Assuming we still need this even with the addition of edx-django-sites-extensions, I think it could probably be moved to a more generic/common location vs. the theming helpers module.